### PR TITLE
Add floor and ceil for fmpq.

### LINKED ISSUE #736: floor and ceil for fmpq
Can we have floor and ceil for fmpq?

### PR DESCRIPTION
**Pull request by wbhart**
Closes #736

### CODE REVIEW

File: src/flint/fmpq.jl [hunk 1]
```diff
@@ -138,6 +138,20 @@ function deepcopy_internal(a::fmpq, dict::IdDict)
...
 
+@doc Markdown.doc"""
+    floor(a::fmpq)
+> Returns the greatest integer that is less than or equal to $a$. The result is
+> returned as a rational with denominator $1$.
+"""
+Base.floor(a::fmpq) = fmpq(fdiv(numerator(a), denominator(a)), 1)
+
+@doc Markdown.doc""";
+    floor(a::fmpq)
```
Comment (rfourquet) [inline src/flint/fmpq.jl]: Adjust definition for `ceil`

Comment (wbhart) [reply src/flint/fmpq.jl]: Done.

Comment (rfourquet) [reply src/flint/fmpq.jl]: It seems you didn't update the signature in the docstring though!
```suggestion
    ceil(a::fmpq)
```

Comment (wbhart) [reply src/flint/fmpq.jl]: Whoops, done. Thanks!

**Pull request merged**